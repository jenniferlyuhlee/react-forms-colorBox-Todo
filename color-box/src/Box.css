.Box-group{
    display: flex;
    align-items: center;
    margin: 10px;
}

.Box-btn{
    background-color: rgba(255, 0, 0, 0.3);
    border-radius: 100%;
    margin-left: 10px;
    font-size: 12px;
    height: 40px;
    width: 40px;
    text-align: center;
}

.Box-btn:hover{
    border-color: red;
}