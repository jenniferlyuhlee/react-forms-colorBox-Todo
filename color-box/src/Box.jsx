import React from "react";
import "./Box.css"
/**
 * Component box
 * Props: color, width, height
 * Inherited func: removeBox - removes box when button clicked
 */
function Box({color, width, height, removeBox}) {
    return (
    <div className="Box-group">
        <div className="Box" style = {{backgroundColor: color, 
                    width: `${width}px`,
                    height: `${height}px`}}>
        </div>
        <button className="Box-btn" onClick ={removeBox}>X</button>
    </div>
    )
}

export default Box;