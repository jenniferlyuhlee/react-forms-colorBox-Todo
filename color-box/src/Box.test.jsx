import React from "react";
import { render } from "@testing-library/react"
import Box from "./Box";

// smoke test
test("it renders without crashing", () => {
    render (<Box color="red" width={100} height={100} />)
});

// snapshot test
test("it matches snapshot", () => {
    const {asFragment} = render(<Box color="red" width={100} height={100} />)
    expect(asFragment()).toMatchSnapshot();
})