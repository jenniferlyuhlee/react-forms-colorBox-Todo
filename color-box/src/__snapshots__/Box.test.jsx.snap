// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box-group"
  >
    <div
      class="Box"
      style="background-color: red; width: 100px; height: 100px;"
    />
    <button
      class="Box-btn"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
